File: ccif/src/java/other/org/himalay/ccif/DaylightSavings_1_1.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:08 EST 2016")

public  class DaylightSavings_1_1 extends BinStruct implements PublicBinMsg {




    // members variables
    // daylightSavingsControl
    public short daylightSavingsControl ;

    public DaylightSavings_1_1 () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize daylightSavingsControl

    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read daylightSavingsControl
        {
            daylightSavingsControl=(short)(istream.readUnsignedByte());
            retVal+=1;
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;



        // write daylightSavingsControl
        ostream.writeByte(daylightSavingsControl);
        retVal +=1;
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("DaylightSavings_1_1\n");
        dc.increaseIndent();
        int retVal= 0;
        // write daylightSavingsControl
        dc.indent();
        dc.getPs().println("daylightSavingsControl="+daylightSavingsControl+"(0x"+ Integer.toHexString(daylightSavingsControl) + ")" );
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for daylightSavingsControl
    //public short getDaylightSavingsControl()
    //{
    //    return daylightSavingsControl ;
    //}


    // Setter for daylightSavingsControl
    //public void setDaylightSavingsControl(short val)
    //{
    //    this.daylightSavingsControl= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/DaylightSavings_1_1.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:18 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "DaylightSavings_1_1.h"

#ifdef org_himalay_ccif_DaylightSavings_1_1__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
DaylightSavings_1_1::DaylightSavings_1_1() {
    references= 0;


    // daylightSavingsControl
    daylightSavingsControl= (ui8)0;
}
// Destructor
DaylightSavings_1_1::~DaylightSavings_1_1() {
    // daylightSavingsControl

}

int DaylightSavings_1_1::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // daylightSavingsControl
    {
        daylightSavingsControl=(istream.readUI8());
        retVal+=1;
    }
    return retVal;
}

int DaylightSavings_1_1::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read daylightSavingsControl
    {
        daylightSavingsControl=(istream.readUI8());
        retVal+=1;
    }

    return retVal;
}

int DaylightSavings_1_1::write(DataOutputStream& ostream) { // throws IOException


    int retVal= 0;
    // write daylightSavingsControl
    ostream.writeUI8(daylightSavingsControl);
    retVal +=1;

    return retVal;
}

int DaylightSavings_1_1::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("DaylightSavings_1_1\n");
    dc.increaseIndent();
    int retVal= 0;
    // write daylightSavingsControl
    dc.indent();
    dc.getPs().print("daylightSavingsControl=");
    dc.getPs().println((long)daylightSavingsControl);
    dc.decreaseIndent();
    return retVal;
}


// Getter for daylightSavingsControl
//ui8 DaylightSavings_1_1::getDaylightSavingsControl()
//{
//return this->daylightSavingsControl;
//}

// Setter for daylightSavingsControl
//void DaylightSavings_1_1::setDaylightSavingsControl(ui8 val)
//{
//this->daylightSavingsControl= val;
//}

int DaylightSavings_1_1::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_DaylightSavings_1_1__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::DaylightSavings_1_1* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::DaylightSavings_1_1* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code